taichi/program/async_utils.cpp

```diff
@@ -157,7 +157,8 @@ TaskMeta *get_task_meta(IRBank *ir_bank, const TaskLaunchRecord &t) {
     if (auto global_store = stmt->cast<GlobalStoreStmt>()) {
       if (auto ptr = global_store->ptr->cast<GlobalPtrStmt>()) {
         for (auto &snode : ptr->snodes.data) {
-          meta.output_states.emplace(snode, AsyncState::Type::value);
+          meta.output_states.insert(
```

Comment:
All changes in the remaining of this file are trivial `AsyncState` constructor replacements.